sympy/core/function.py

```diff
@@ -109,6 +109,11 @@ def _getnargs(cls):
             return _getnargs_old(cls.eval)
         else:
             return _getnargs_new(cls.eval)
+    elif inspect.isfunction(cls):
+        if sys.version_info < (3, ):
```

Comment:
You can import `PY3` from `sympy.core.compatibility` and use here (although the if should be reversed then...).